bpi.module

```diff
@@ -857,3 +555,207 @@ function bpi_error_message($exception, $level) {
   // Send the exception to logger.
   watchdog_exception('bpi', $exception);
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Add custom submit handler for showing BPI specific fields and saving BPI
+ * data in db and add validate handler, which will add our own submit handler.
+ */
+function bpi_form_ding_news_node_form_alter(&$form, &$form_state, $form_id) {
```

Comment:
Will language always be LANGUAGE_NONE in the workflow fields below, should the language come from the node being edited ?